### DISCUSSION TITLE
Is chmod and chown applied to a file before moving from temporary location to final destination or after?

### DISCUSSION THREAD

**Discussion by thorewi** [RsyncProject/rsync#289] [Q&A] [answered]
Hello,

is chmod and chown applied to a file before moving from temporary location to final destination or after? And if after, shouldn't it be before to get atomicity? Thank you.

**Comment by WayneD** [comment, marked as answer]
The temporary file is finalized before moving it into place.  It is also chowned prior to the changing of the mode from its more restricted temp-file mode to the final mode.
